core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -209,6 +210,7 @@ static HttpResponse of(HttpHeaders headers, HttpData content, HttpHeaders traili
         }
 
         if (isContentAlwaysEmptyWithValidation(status, content, trailingHeaders)) {
+            ReferenceCountUtil.safeRelease(content);
```

Comment:
Do you still want to keep these checks with the other change? I agree that it is pretty hairy (we'll want to add below too I think for the branches triggered by `content.isEmpty()`

Reply:
Oh nevermind, these are cases where the buf might not be empty, got it